wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -766,6 +949,31 @@ instance Arbitrary SyncState where
                     , pure Synced
                     ]
 
+-- | 'Wallet' type.
+data WalletType
+    = WalletRegular
+    -- ^ Regular Cardano wallet.
+    | WalletExternal
+    -- ^ External wallet (mobile app or hardware wallet).
+    deriving (Bounded, Enum, Eq, Ord, Show, Generic)
+
+instance Arbitrary WalletType where
+    arbitrary = elements [minBound .. maxBound]
+
+-- Drops the @Wallet@ prefix.
+deriveJSON Serokell.defaultOptions { A.constructorTagModifier = drop 6 . map C.toLower
+                                   } ''WalletType
+
+instance ToSchema WalletType where
+    declareNamedSchema _ = do
+        pure $ NamedSchema (Just "WalletType") $ mempty
+            & type_ .~ SwaggerString
+            & enum_ ?~ ["regular", "external"]
```

Comment:
Minor tips but:

```hs
import Cardano.Wallet.API.V1.Generic (gconsNames)

...

instance ToSchema WalletType where
    declareNamedSchema _ = do
        pure $ NamedSchema (Just "WalletType") $ mempty
            & type_ .~ SwaggerString
            & enum_ ?~ map constructorTagModifier (gconsNames (Proxy @WalletType))
```

Reply:
It's more complex. Should we prefer such a code?

Reply:
A bit more complex, but it leverages generics so if you add or change a `WalletType`, you get the schema updated for free. In the end, we're also doing quickcheck verification on the ToSchema / ToJSON relationship, so if you'd forget to update the schema, the tests will catch it (hopefully). 

Still, it's rather simple IMO :smile: 
Your call.